### PR TITLE
Implementing ArtFileManager

### PR DESCRIPTION
**Pull request by AnnaBanani**
Implementing ArtFileManager

### CODE REVIEW

File: TheMET/TheMET/DataModels/Art.swift [hunk 1]
```diff
@@ -7,51 +7,52 @@
...
-        self.objectDate = objectDate
-        self.medium = medium
-        self.creditLine = creditLine
-        self.country = country
-        self.classification = classification
-    }
-    
-}
+typealias Art = ObjectResponse
+//class Art {
```
Comment (Ssimboss) [inline TheMET/TheMET/DataModels/Art.swift:11]: Please remove all the commented lines in this file

File: TheMET/TheMET/Network/ArtFileManager.swift [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+           return nil
+       }
+    }
+    
+    func write(art: Art) {
+        let jsonEncoder = JSONEncoder()
+        guard let jsonData = try? jsonEncoder.encode(art) else {
+            return
+        }
+        guard String(data: jsonData, encoding: .utf8) != nil else {
```
Comment (Ssimboss) [inline TheMET/TheMET/Network/ArtFileManager.swift:29]: Why do you do this check?

File: TheMET/TheMET/Network/ArtFileManager.swift [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                    return
+                }
+            }
+            fileManager.createFile(atPath: fileURL.path(), contents: jsonData)
+        }
+    }
+    
+    func readArt(id: Int, completion: @escaping (Art?) -> Void) {
+        let fileManager = FileManager.default
+        guard fileManager.fileExists(atPath: self.folderURL.path(percentEncoded: false)) else {
```
Comment (Ssimboss) [inline TheMET/TheMET/Network/ArtFileManager.swift:48]: The folder check is unnecessary as you check the existence of the file a few lines later

File: TheMET/TheMET/Network/ArtFileManager.swift [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+               let loadedArt = try? decoder.decode(Art.self, from: artData) {
+                executeComplitionOnMain(loadedArt)
+            } else {
+                executeComplitionOnMain(nil)
+            }
+        }
+    }
+    
+    func readArtIds(completion: @escaping ([Int]) -> Void) {
+        var filesIds:[Int] = []
```
Comment (Ssimboss) [inline TheMET/TheMET/Network/ArtFileManager.swift:74]: These are `artIds`, not file-identifiers

File: TheMET/TheMET/Network/ArtFileManager.swift [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            return
+        }
+        DispatchQueue.global().async {
+            do {
+                let directoryContents = try FileManager.default.contentsOfDirectory(at: self.folderURL, includingPropertiesForKeys: nil, options: [])
+                let folderURLString = self.folderURL.path(percentEncoded: false)
+                let urlStringSuffix: String = ".string"
+                for file in directoryContents {
+                    let contentString = file.path(percentEncoded: false)
+                    var modifiedContentString = contentString.replacingOccurrences(of: folderURLString, with: "")
```
Comment (Ssimboss) [inline TheMET/TheMET/Network/ArtFileManager.swift:87]: There is a better way to get a filename from URL. check for [pathComponents](https://developer.apple.com/documentation/foundation/url/1780324-pathcomponents) or even [lastPathComponent](https://developer.apple.com/documentation/foundation/url/1780317-lastpathcomponent)

File: TheMET/TheMET/Network/ArtFileManager.swift [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+                    modifiedContentString = modifiedContentString.replacingOccurrences(of: urlStringSuffix, with: "")
+                    if let id: Int = Int(modifiedContentString) {
+                        filesIds.append(id)
+                    }
+                }
+                completion(filesIds)
```
Comment (Ssimboss) [inline TheMET/TheMET/Network/ArtFileManager.swift:93]: The completion is executed of DispatchQueue.global

**Review by Ssimboss — CHANGES_REQUESTED** [review]

File: TheMET/TheMET/Network/ArtFileManager.swift (hunk 3 above)
Comment (AnnaBanani) [reply TheMET/TheMET/Network/ArtFileManager.swift:48]: fixed

File: TheMET/TheMET/Network/ArtFileManager.swift (hunk 5 above)
Comment (AnnaBanani) [reply TheMET/TheMET/Network/ArtFileManager.swift:87]: fixed

File: TheMET/TheMET/Network/ArtFileManager.swift (hunk 6 above)
Comment (AnnaBanani) [reply TheMET/TheMET/Network/ArtFileManager.swift:93]: fixed

File: TheMET/TheMET/Network/ArtFileManager.swift (hunk 2 above)
Comment (AnnaBanani) [reply TheMET/TheMET/Network/ArtFileManager.swift:29]: fixed

**Review by Ssimboss — APPROVED** [review]

**Pull request merged**